### PR TITLE
Refactor gnps classes

### PR DESCRIPTION
**Pull request by CunliangGeng**
This PR intends to check GNPS data, refactor classes and functions on downloading, extracting and loading GNPS data, add detailed docstring to explain how GNPS data is processed, and update unit tests for all types of GNPS data. 

**👉 It's easier to review the code commit by commit or class by class.**
Though this PR has so many commits, actually the changes on GNPS classes followed the similar pattern.

### Major changes

- Removed dependency `requests` and only use `httpx` to handle http requests
- Added dependency `pytemoics` to parse MGF file 
- Added the missing GNPS workflow (`SNETS-V2`) to enum `GNPSFormat` 
- Refactored all classes and functions in the module `gnps`, including `GNPSDownloader`, `GNPSExtractor`, `GNPSAnnotationLoader`, `GNPSFileMappingLoader`, `GNPSMolecularFamilyLoader`, `GNPSSpectrumLoader`, and functions from `gnps_format.py`
	- Updated relevant download/extract/load logics to consider all GNPS workflows in `GNPSFormat`
	- Added data validation before loading
	- Added detailed docstring for each GNPS data type
	- Fixed some bugs or invalid URLs
- Refactored all unit tests relevant to the `gnps` module
	- Replaced minimal working test data with complete GNPS data (zip file)
	- Restructured and refactored module-level fixtures to make all tests consistent with each other and easier to understand
	- Refactored all tests to test all types of GNPS workflows
- Added, removed or updated some util functions in `utils.py`

### Expected failed tests
- all tests in `pairedomics` folder
- `test_loader.py`

### CODE REVIEW

File: src/nplinker/metabolomics/gnps/gnps_format.py [hunk 1]
```diff
@@ -88,37 +88,41 @@ def gnps_format_from_task_id(task_id: str) -> GNPSFormat:
...
         return GNPSFormat.SNETSV2
-    if "METABOLOMICS-SNETS" in file.name:
+    if GNPSFormat.SNETS.value in file.name:
         return GNPSFormat.SNETS
 
     # Guess the format from the names of the files in the zip file
     with zipfile.ZipFile(file) as archive:
         filenames = archive.namelist()
-    if any("FEATURE-BASED-MOLECULAR-NETWORKING" in x for x in filenames):
+    if any(GNPSFormat.FBMN.value in x for x in filenames):
```
Comment (gcroci2) [inline src/nplinker/metabolomics/gnps/gnps_format.py:92]: For all these if statements, why is it not needed to verify all of the filenames, but at least one is sufficient?

File: src/nplinker/metabolomics/gnps/gnps_format.py [hunk 2]
```diff
@@ -5,9 +5,10 @@
 import zipfile
 from bs4 import BeautifulSoup
 from bs4 import Tag
-import requests
+import httpx
```
Comment (gcroci2) [inline src/nplinker/metabolomics/gnps/gnps_format.py:8]: Googling httpx vs requests I've read that they have similar performances. I was wondering why the change?

File: src/nplinker/utils.py [hunk 3]
```diff
@@ -345,36 +346,41 @@ def extract_archive(from_path: str | PathLike,
...
+        from_path: Path to the file to be extracted.
+        extract_root: Path to the directory the file will be extracted to.
             If omitted, the directory of the archive file is used.
-        remove_finished (bool): If ``True``, remove the file after the extraction.
+        members: Optional selection of members to extract. If not specified,
+            all members are extracted.
+            Memers must be a subset of the list returned by
+            - ``zipfile.ZipFile.namelist()`` or a list of strings for zip file
+            - ``tarfile.TarFile.getmembers()`` for tar file
+        remove_finished: If ``True``, remove the file after the extraction.
```
Comment (gcroci2) [inline src/nplinker/utils.py]: General question, why do we use double ` in the doc strings?

File: src/nplinker/metabolomics/gnps/gnps_annotation_loader.py [hunk 4]
```diff
@@ -1,42 +1,134 @@
...
+                        f"but got '{header}'.")
 
-        Examples:
-            >>> print(loader.annotations()[100])
-            """
-        return self._annotations
+        # validate that "#Scan#" must be unique
+        with open(self._file, mode='rt') as f:
+            reader = csv.DictReader(f, delimiter='\t')
+            ids = []
```
Comment (gcroci2) [inline src/nplinker/metabolomics/gnps/gnps_annotation_loader.py]: What is the purpose of this list?

**Review by gcroci2 — APPROVED** [review]
Minor comments :)

File: src/nplinker/metabolomics/gnps/gnps_format.py (hunk 1 above)
Comment (CunliangGeng) [reply src/nplinker/metabolomics/gnps/gnps_format.py:92]: There is actually only one file in GNPS output that match the name pattern.

File: src/nplinker/metabolomics/gnps/gnps_format.py (hunk 2 above)
Comment (CunliangGeng) [reply src/nplinker/metabolomics/gnps/gnps_format.py:8]: They do almost the same things but both are dependencies of NPLinker. Keeping only one and sticking to it will make life easier ;-)   
Why keeping `httpx` then? It's kind of `requests+`, having all features of `requests` and more. One of the important new features is asynchronous request, which helps scale the downloading. Probably we'll use it later.

File: src/nplinker/utils.py (hunk 3 above)
Comment (CunliangGeng) [reply src/nplinker/utils.py]: No idea. Changed to single backtick.

File: src/nplinker/metabolomics/gnps/gnps_annotation_loader.py (hunk 4 above)
Comment (CunliangGeng) [reply src/nplinker/metabolomics/gnps/gnps_annotation_loader.py]: It acts as a temporary container. This validation of unique ids is not intuitive, updated it in the new commit.

**Review by gcroci2 — APPROVED** [review]
